### PR TITLE
mix - Logs window

### PR DESCRIPTION
**Pull request by yann300**
- logs pane below the status pane
 - retrieve javascript message

### CODE REVIEW

File: mix/CodeEditorExtensionManager.cpp [hunk 1]
```diff
@@ -53,10 +53,10 @@ void CodeEditorExtensionManager::loadEditor(QQuickItem* _editor)
 
 void CodeEditorExtensionManager::initExtensions()
 {
-	std::shared_ptr<StatusPane> output = std::make_shared<StatusPane>(m_appContext);
-	QObject::connect(m_appContext->codeModel(), &CodeModel::compilationComplete, this, &CodeEditorExtensionManager::applyCodeHighlight);
+	//std::shared_ptr<StatusPane> output = std::make_shared<StatusPane>(m_appContext);
```
Comment (subtly) [inline mix/CodeEditorExtensionManager.cpp]: Can these lines be removed?

File: mix/qml/MainContent.qml [hunk 2]
```diff
@@ -102,7 +102,7 @@ Rectangle {
 	}
 
 	CodeEditorExtensionManager {
-		headerView: headerPaneTabs;
+		//headerView: headerPaneTabs;
```
Comment (subtly) [inline mix/qml/MainContent.qml]: Can this be removed?

File: mix/qml/StatusPane.qml [hunk 3]
```diff
@@ -133,6 +177,60 @@ Rectangle {
...
+				{
+					statusContainer.state = "logsOpened";
+					logsContainer.state = "opened";
+				}
+			}
+
+			id: logsContainer
+			width: 1000
+			height: 0
+			//anchors.topMargin:
```
Comment (subtly) [inline mix/qml/StatusPane.qml]: Safe to remove?

File: mix/qml/StatusPane.qml [hunk 4]
```diff
@@ -150,7 +247,9 @@ Rectangle {
 		id: displayLogAction
 		tooltip: qsTr("Display Log")
 		onTriggered: {
-			mainContent.displayCompilationErrorIfAny();
+			logsContainer.toggle();
+			//if (status.state === "error" && logPane.front().type === "run")
```
Comment (subtly) [inline mix/qml/StatusPane.qml]: Is there a debug state that can be accessed to enable this code?

**Comment by subtly** [comment]
Ah, sorry. Didn't see the 'pleasereview' tag hadn't been set yet.

File: mix/AppContext.cpp [hunk 5]
```diff
@@ -37,6 +38,7 @@
 #include "QVariableDefinition.h"
 #include "HttpServer.h"
 #include "AppContext.h"
+#include "Sortfilterproxymodel.h"
```
Comment (arkpar) [inline mix/AppContext.cpp]: should be SortFilterProxyModel.h

File: mix/AppContext.cpp [hunk 6]
```diff
@@ -22,6 +22,7 @@
  * - KeyEventManager
  */
 
+#include <QSortFilterProxyModel>
```
Comment (arkpar) [inline mix/AppContext.cpp]: This is probably not required

File: mix/Sortfilterproxymodel.cpp [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+#include "Sortfilterproxymodel.h"
+#include <QtDebug>
+#include <QtQml>
+
+using namespace dev::mix;
+
+SortFilterProxyModel::SortFilterProxyModel(QObject* _parent) : QSortFilterProxyModel(_parent)
+{
+	connect(this, SIGNAL(rowsInserted(QModelIndex,int,int)), this, SIGNAL(countChanged()));
```
Comment (arkpar) [inline mix/Sortfilterproxymodel.cpp]: spacing, + better use function addresses for connect instead of macros

File: mix/Sortfilterproxymodel.cpp [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+QObject* SortFilterProxyModel::source() const
+{
+	return sourceModel();
+}
+
+void SortFilterProxyModel::setSource(QObject* _source)
+{
+	setSourceModel(qobject_cast<QAbstractItemModel *>(_source));
```
Comment (arkpar) [inline mix/Sortfilterproxymodel.cpp]: spacing

File: mix/Sortfilterproxymodel.cpp [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+void SortFilterProxyModel::setFilterSyntax(SortFilterProxyModel::FilterSyntax _syntax)
+{
+	setFilterRegExp(QRegExp(filterString(), filterCaseSensitivity(), static_cast<QRegExp::PatternSyntax>(_syntax)));
+}
+
+QJSValue SortFilterProxyModel::get(int _idx) const
+{
+	QJSEngine *engine = qmlEngine(this);
+	QJSValue value = engine->newObject();
+	if (_idx >= 0 && _idx < count()) {
```
Comment (arkpar) [inline mix/Sortfilterproxymodel.cpp]: braces styling in this and the following function

File: mix/Sortfilterproxymodel.cpp [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+QHash<int, QByteArray> SortFilterProxyModel::roleNames() const
+{
+	if (QAbstractItemModel* source = sourceModel())
+		return source->roleNames();
+	return QHash<int, QByteArray>();
+}
+
+bool SortFilterProxyModel::filterAcceptsRow(int _sourceRow, QModelIndex const& _sourceParent) const
+{
+
```
Comment (arkpar) [inline mix/Sortfilterproxymodel.cpp]: extra line

File: mix/Sortfilterproxymodel.cpp [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+	QRegExp rx = filterRegExp();
+	QAbstractItemModel *model = sourceModel();
+	QModelIndex sourceIndex = model->index(_sourceRow, 0, _sourceParent);
+		if (!sourceIndex.isValid())
```
Comment (arkpar) [inline mix/Sortfilterproxymodel.cpp]: indentation

**Pull request merged**